### PR TITLE
Move ModificationExecutors to new interface

### PR DESCRIPTION
**Pull request by markuspf**
Team Gordo internal PR

### CODE REVIEW

**Comment by hkernbach** [comment]
http://jenkins01:8080/view/PR/job/arangodb-matrix-pr/8808/

File: arangod/Aql/ModificationExecutor.cpp [hunk 1]
```diff
@@ -157,53 +154,89 @@ void ModificationExecutor<FetcherType, ModifierType>::doOutput(OutputAqlItemRow&
...
   }
 
-  TRI_IF_FAILURE("ModificationBlock::getSome") {
-    THROW_ARANGO_EXCEPTION(TRI_ERROR_DEBUG);
-  }
+  if (_modifier.nrOfOperations() > 0) {
+    _modifier.transact();
 
-  std::tie(_lastState, stats) = doCollect(maxOutputs);
+    if (_infos._doCount) {
```
Comment (hkernbach) [inline arangod/Aql/ModificationExecutor.cpp]: Why is that not included in doOutput anymore?

File: arangod/Aql/ModificationExecutor.cpp [hunk 2]
```diff
@@ -157,53 +154,89 @@ void ModificationExecutor<FetcherType, ModifierType>::doOutput(OutputAqlItemRow&
...
+  if (_modifier.nrOfOperations() > 0) {
+    _modifier.transact();
 
-  // If the query is silent, there is no way to relate
-  // the results slice contents and the submitted documents
-  // If the query is *not* silent, we should get one result
-  // for every document.
-  // Yes. Really.
-  TRI_ASSERT(_infos._options.silent || _modifier.nrOfDocuments() == _modifier.nrOfResults());
+    if (_infos._doCount) {
```
Comment (hkernbach) [inline arangod/Aql/ModificationExecutor.cpp]: Why is that not included in doOutput anymore?

**Comment by hkernbach** [comment]
http://jenkins01:8080/view/PR/job/arangodb-matrix-pr/8810/

File: arangod/Aql/ModificationExecutor.cpp (hunk 2 above)
Comment (mchacki) [reply arangod/Aql/ModificationExecutor.cpp]: Because we need it in the skip as well.
The doOutput does not need the Stats any more to make it clearer

**Comment by mchacki** [comment]
Red Jenkins will be fixed in base-branch, consolidating all now

**Pull request merged**